experiment/measurer/detailed_coverage_data_utils.py

```diff
@@ -31,123 +31,68 @@ class DetailedCoverageData:  # pylint: disable=too-many-instance-attributes
-        resulting CSV file."""
-        try:
-            # Drop duplicates but with different timestamps in segment data.
-            self.segment_df = self.segment_df.sort_values(by=['time'])
-            self.segment_df = self.segment_df.drop_duplicates(
-                subset=self.segment_df.columns.difference(['time']),
-                keep='first')
-            self.name_df = self.name_df.drop_duplicates(keep='first')
-
-        except (ValueError, KeyError, IndexError):
-            coverage_utils.logger.error(
-                'Error occurred when removing duplicates.')
-
-    def generate_csv_files(self):
+
+        insert_series = pd.Series(segment_entry, index=self.segment_df.columns)
+        self.segment_df = self.segment_df.append(insert_series,
+                                                 ignore_index=True)
+
+    def generate_csv_files(self, trial_measurement_dir, trial_id, cycle):
         """Generates three compressed CSV files containing coverage information
         for all fuzzers, benchmarks, and trials. To maintain a small file size,
         all strings, such as file and function names, are referenced by id and
         resolved in 'names.csv'."""
 
-        # Clean and prune experiment-specific data frames.
-        self.remove_redundant_entries()
-
         # Write CSV files to filestore.
-        def csv_filestore_helper(file_name, df):
+        def csv_filestore_helper(file_name, df, file_type):
             """Helper method for storing csv files in filestore."""
-            src = os.path.join(coverage_utils.get_coverage_info_dir(), 'data',
-                               file_name)
-            dst = exp_path.filestore(src)
+
+            if file_type == 'function':
+                src = os.path.join(trial_measurement_dir, 'functions_coverage',
+                                   file_name)
+            else:
+                src = os.path.join(trial_measurement_dir, 'segments_coverage',
+                                   file_name)
+
             df.to_csv(src, index=False, compression='infer')
-            filestore_utils.cp(src, dst)
+            # the following commented lines lets us copy the file from instance
+            # file store to the local file store or the bucket.
+            #dst = exp_path.filestore(src)
+            #filestore_utils.cp(src, dst)
 
-        csv_filestore_helper('functions.csv.gz', self.function_df)
-        csv_filestore_helper('segments.csv.gz', self.segment_df)
-        csv_filestore_helper('names.csv.gz', self.name_df)
+        csv_filestore_helper(
+            'functions_{trial_id}_{cycle}.csv.gz'.format(trial_id=trial_id,
+                                                         cycle=cycle),
+            self.function_df, 'function')
+        csv_filestore_helper(
+            'segments_{trial_id}_{cycle}.csv.gz'.format(trial_id=trial_id,
+                                                        cycle=cycle),
+            self.segment_df, 'segment')
```

Comment:
All of this is being taken care of by the state file. Why are you still trying to persist trial and cycle specific information yourself?